src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -651,18 +652,30 @@ public function load(array $configs, ContainerBuilder $container)
         $container->registerAttributeForAutoconfiguration(AsController::class, static function (ChildDefinition $definition, AsController $attribute): void {
             $definition->addTag('controller.service_arguments');
         });
-        $container->registerAttributeForAutoconfiguration(AsMessageHandler::class, static function (ChildDefinition $definition, AsMessageHandler $attribute, \ReflectionClass|\ReflectionMethod $reflector): void {
-            $tagAttributes = get_object_vars($attribute);
-            $tagAttributes['from_transport'] = $tagAttributes['fromTransport'];
-            unset($tagAttributes['fromTransport']);
-            if ($reflector instanceof \ReflectionMethod) {
-                if (isset($tagAttributes['method'])) {
-                    throw new LogicException(sprintf('AsMessageHandler attribute cannot declare a method on "%s::%s()".', $reflector->class, $reflector->name));
+
+        if (class_exists(SerializedMessageStamp::class)) {
```

Comment:
alternatively, we would have to conflict with `symfony/messenger` before 6.1

Reply:
As 6.1 is already released, I think having a condition is ok. On 6.2, we can then remove the condition and add a conflict.

Reply:
That was my reasoning too for choosing this solution. Will be simplified on `6.2` with #47237.